File: test_data/code/python/1678-goal-parser-interpretation.py
class Solution:
    def interpret(self, command: str) -> str:
        result = ""
        i = 0
        while i < len(command):
            if command[i] == "G":
                result += command[i]
                i += 1
            elif command[i:i+2] == "()":
                result += "o"
                i += 2
            elif command[i:i+3] == "(al)":
                result += "al"
                i += 3
        return result

File: updated_code/code/java/1678-goal-parser-interpretation.java
package orggoals.parser;

import java.utilStringBuilder;

public class GoalParser {

    public String interpret(String command) {
        StringBuilder res = new StringBuilder();
        for(int i = 0; i < command.length(); i++) {
            if(command.charAt(i) == 'G') {
                res.append('G');
            } else if(command.charAt(i) == '(') {
                int j = i + 1;
                while(j < command.length() && command.charAt(j) != ')') {
                    j++;
                }
                if(j < command.length() && command.substring(i, j + 1).equals("(al)")) {
                    res.append("al");
                    i = j;
                } else {
                    res.append('o');
                }
            }
        }
        return res.toString();
    }
}